src/app/src/components/FilterSidebarFacilitiesTab.jsx

```diff
@@ -222,11 +223,23 @@ function FilterSidebarFacilitiesTab({
                             />
                         </div>
                     ) : (
-                        <DownloadFacilitiesButton
-                            setLoginRequiredDialogIsOpen={
-                                setLoginRequiredDialogIsOpen
+                        <FeatureFlag
+                            flag={ALLOW_LARGE_DOWNLOADS}
+                            alternative={
+                                <DownloadFacilitiesButton
+                                    disabled={facilitiesCount >= 10000}
```

Comment:
Can this check be made into a utility function and/or use a constant for the 10,000 (for clarity/DRYness)?